homeassistant/data_entry_flow.py

```diff
@@ -311,7 +321,60 @@ async def async_configure(
         if (
             data_schema := cur_step.get("data_schema")
         ) is not None and user_input is not None:
-            user_input = data_schema(user_input)
+            try:
+                user_input = data_schema(user_input)
+            except vol.Invalid as ex:
+                raised_errors = [ex]
+                if isinstance(ex, vol.MultipleInvalid):
+                    raised_errors = ex.errors
+
+                schema_errors: dict[str, Any] = {}
+                for error in raised_errors:
+                    if error.path:
+                        current_path_schema = data_schema.schema
+                        current_path_errors = schema_errors
+                        for path_part in error.path[:-1]:
+                            if path_part not in current_path_schema:
+                                current_path_schema = None
+                                break
+                            current_path_schema = current_path_schema[path_part]
+                            current_path_errors = current_path_errors.setdefault(
+                                path_part, {}
+                            )
+                        if (
+                            current_path_schema
+                            and error.path[-1] in current_path_schema
+                        ):
+                            current_path_errors[error.path[-1]] = error.error_message
+                            continue
+
+                    # If we get here, we could not identify the path of the error.
+                    schema_errors.get("base", []).append(str(error))
+
+                class SchemaInvalidError(
```

Comment:
We create the error class dynamically so we have the same type as the original error.
Motivation is that many tests expect a `vol.Invalid` error and the flow manager is not only used for config flows.
I'm fine to removing this "backwards" compatible code, if we are fine to change it